src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -78,13 +78,13 @@ func (a *App) onBootstrapWorkerNode(ctx context.Context, s *state.State, encoded
 			log.Info("Cleaning up")
 			for i := len(cleanups) - 1; i >= 0; i-- {
 				// run cleanup functions in reverse order
-				if err := cleanups[i](s.Context); err != nil {
+				if err := cleanups[i](ctx); err != nil {
 					log.Error(err, fmt.Sprintf("Cleanup hook %d/%d failed", i, len(cleanups)))
 				}
 			}
 			log.Info("All cleanup hooks finished, resetting microcluster state")
 
-			if err := a.client.ResetClusterMember(s.Context, s.Name(), true); err != nil {
+			if err := a.client.ResetClusterMember(ctx, s.Name(), true); err != nil {
```

Comment:
this is not right, as the request context is cancelled after we've sent the response. see my other comment about a `.Context()` that gives us the base context.

Reply:
https://github.com/canonical/microcluster/pull/193 updates the ctx to be the microcluster daemon context instead of the request context.

Reply:
obsolete